test/axes/baseAxisTests.ts

```diff
@@ -206,4 +206,593 @@ describe("BaseAxis", () => {
+          axis.annotatedTicks(annotatedTicks);
+          axis.annotationsEnabled(true);
+
+          let svg = TestMethods.generateSVG(300, 300);
+          axis.renderTo(svg);
+
+          let annotationRects = axis.content().selectAll(".annotation-rect");
+
+          axis.content().selectAll(".annotation-circle").each(function (d, i) {
+            let annotationCircle = d3.select(this);
+            let correspondingRect = d3.select(annotationRects[0][i]);
+            assert.closeTo(TestMethods.numAttr(annotationCircle, "cx"), 0, window.Pixel_CloseTo_Requirement, "circle at the left");
+            assert.closeTo(TestMethods.numAttr(annotationCircle, "cy"),
+              TestMethods.numAttr(correspondingRect, "y"), window.Pixel_CloseTo_Requirement, "circle at tick scaled y position");
+          });
+
+          axis.content().selectAll(".annotation-line").each(function (d, i) {
+            let annotationLine = d3.select(this);
+            let correspondingRect = d3.select(annotationRects[0][i]);
+            assert.closeTo(TestMethods.numAttr(annotationLine, "x1"), 0, window.Pixel_CloseTo_Requirement, "line starts at the left");
+            assert.closeTo(TestMethods.numAttr(annotationLine, "x2"),
+              TestMethods.numAttr(correspondingRect, "x"), window.Pixel_CloseTo_Requirement, "line ends at the rect");
+            assert.closeTo(TestMethods.numAttr(annotationLine, "y1"),
+              TestMethods.numAttr(correspondingRect, "y"), window.Pixel_CloseTo_Requirement, "line at tick scaled y position");
+            assert.closeTo(TestMethods.numAttr(annotationLine, "y2"),
+              TestMethods.numAttr(correspondingRect, "y"), window.Pixel_CloseTo_Requirement, "line at tick scaled y position");
+          });
+
+          axis.content().selectAll(".annotation-label text").each(function (d, i) {
+            let annotationLabel = d3.select(this);
+            let surroundingRect = d3.select(annotationRects[0][i]);
+            TestMethods.assertBBoxInclusion(surroundingRect, annotationLabel);
+          });
+          svg.remove();
+        });
+      });
+
+      describe("annotation rectangle placement", () => {
+        let scale: Plottable.Scales.Linear;
+        let annotatedTicks: number[];
+
+        beforeEach(() => {
+          scale = new Plottable.Scales.Linear();
+          annotatedTicks = [50, 100, 150];
+          scale.domain([0, 300]);
+        });
+
+        describe("scaled position along axis", () => {
+          it("places the rectangle at the scaled x position for bottom orientation", () => {
+            let axis = new Plottable.Axis(scale, "bottom");
+            axis.annotatedTicks(annotatedTicks);
+            axis.annotationsEnabled(true);
+
+            let svg = TestMethods.generateSVG(300, 300);
+            axis.renderTo(svg);
+
+            let annotationRects = axis.content().selectAll(".annotation-rect");
+            annotationRects.each(function (d) {
```

Comment:
Can you make sure to check the size of the selection, as per [this best practice](https://github.com/palantir/plottable/wiki/Testing-best-practices#always-check-the-size-of-a-d3-selection)?

Here and elsewhere there are `each()` loops.